### PR TITLE
Implement support for multi-output recipes

### LINKED ISSUE #593: Missing support for multi-output recipes
~~This isn't a major problem at the moment, since the few recipes that do have multiple outputs aren't implemented, however this should still be resolved as soon as possible.~~ Now it is!

### PR DESCRIPTION
**Pull request by JustTalDevelops**
Resolves #593.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -2,15 +2,20 @@ package main
...
 	"github.com/df-mc/dragonfly/server/player/chat"
+	"github.com/df-mc/dragonfly/server/world"
 	"github.com/pelletier/go-toml"
 	"github.com/sirupsen/logrus"
 	"io/ioutil"
 	"os"
 )
 
 func main() {
+	df_commands.RegisterAll()
```
Comment (Flonja) [inline main.go:17]: Obviously not part of the pr

File: server/session/handler_item_stack_request.go [hunk 2]
```diff
@@ -300,6 +337,7 @@ func (h *ItemStackRequestHandler) verifySlot(slot protocol.StackRequestSlotInfo,
 	// The client seems to send negative stack network IDs for predictions, which we can ignore. We'll simply
 	// override this network ID later.
 	if id := item_id(i); id != clientID {
+		fmt.Printf("%v\n", i)
```
Comment (Flonja) [inline server/session/handler_item_stack_request.go:340]: Remove this debug line

File: main.go [hunk 3]
```diff
@@ -28,7 +33,9 @@ func main() {
 		log.Fatalln(err)
 	}
 
-	for srv.Accept(nil) {
+	for srv.Accept(func(p *player.Player) {
```
Comment (Flonja) [inline main.go:36]: This also isn't part of this pr

**Review by Flonja — CHANGES_REQUESTED** [review]
Small easy changes. I haven't looked into the specifics since I don't know about them.

**Review by DaPigGuy — APPROVED** [review]

File: server/session/handler_item_stack_request.go [hunk 4]
```diff
@@ -269,6 +275,39 @@ func (h *ItemStackRequestHandler) handleMineBlock(a *protocol.MineBlockStackRequ
 	return nil
 }
 
+// handleCreate handles the creation of a pending result through a craft.
```
Comment (Sandertv) [inline server/session/handler_item_stack_request.go:278]: I would document this more thoroughly. What does the `CreateStackRequestAction` do here exactly? I have a bit of a hard time understanding what's happening here.

**Review by Sandertv — COMMENTED** [review]
There's quite some changes in here and I'm having a bit of a hard time understanding exactly what the relevant ones are. I would try documenting those a bit more thoroughly.

**Comment by Sandertv** [comment]
@JustTalDevelops please see my comment.

File: server/session/handler_item_stack_request.go (hunk 4 above)
Comment (JustTalDevelops) [reply server/session/handler_item_stack_request.go:278]: It's a stack request action sent by the client when a recipe outputs more than one item. It contains a result slot, which should map to one of the output items and the server should create the relevant output as usual.

Comment (Sandertv) [reply server/session/handler_item_stack_request.go:278]: You should document that :wink:

Comment (JustTalDevelops) [reply server/session/handler_item_stack_request.go]: Updated the documentation there.

**Review by Sandertv — APPROVED** [review]
This looks good to me. Feel free to merge if it's been tested thoroughly enough.
